hydro_lang/src/ir.rs

```diff
@@ -28,6 +28,47 @@ use crate::backtrace::Backtrace;
 use crate::deploy::{Deploy, RegisterPort};
 use crate::location::LocationId;
 
+/// Represents the kind of stream/collection type for metadata
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
+pub enum StreamKind {
+    /// Regular stream - ordering/retries determined by type parameters
+    Stream,
+    /// Keyed stream - ordering/retries determined by type parameters  
+    KeyedStream,
+    /// Singleton collection
+    Singleton,
+    /// Optional collection
+    Optional,
+}
+
+/// Mirror enum for ordering metadata, corresponding to phantom types
```

Comment:
These are no longer needed since the `StreamKind` was simplified?

Reply:
addressed